src/basic/path-util.c

```diff
@@ -639,21 +639,30 @@ static int check_x_access(const char *path, int *ret_fd) {
         return 0;
 }
 
-int find_executable_full(const char *name, bool use_path_envvar, char **ret_filename, int *ret_fd) {
+int find_executable_full(const char *name, const char *root, bool use_path_envvar, char **ret_filename, int *ret_fd) {
         int last_error, r;
         const char *p = NULL;
 
         assert(name);
 
         if (is_path(name)) {
                 _cleanup_close_ int fd = -1;
+                _cleanup_free_ char *path_name = NULL;
 
-                r = check_x_access(name, ret_fd ? &fd : NULL);
+                r = chase_symlinks(name,
```

Comment:
Resolving the symlinks changed the behavior of a few callers in a way that alters the output - that's why some tests are failing. For now, it's safer to try and keep the existing behavior the same as it might be a soft-API.

Hence, please do something like:

```
if (root) {
  chase_symlinks...
```

Reply:
also given it's slightly more complex, please split this commit in two - first one to change src/basic|core|test (and passing NULL in verify_executable's call to `find_executable_full`), and the analyze changes to actually use `root` by themselves.